src/lib/mina/v2/core.ts

```diff
@@ -95,6 +95,44 @@ class TokenId {
   }
 
   static MINA: TokenId = new TokenId(new Field(1));
+
+  /**
+   * Derives a TokenId from a token owner's public key and a parent token ID.
+   * This implementation exactly matches the OCaml code in Mina Protocol:
+   * https://github.com/MinaProtocol/mina/blob/2b0f60cf908bf8ed74de98fcc5442d3947f89d8e/src/lib/mina_base/account_id.ml#L133
+   * 
+   * @param tokenOwner - The public key of the token owner.
+   * @param parentTokenId - The parent token ID (defaults to MINA token ID).
+   * @returns A new TokenId.
+   */
+  static derive(tokenOwner: PublicKey, parentTokenId: Field = new Field(1)): TokenId {
+    // In Mina protocol, they create an "owner" object which is an AccountId containing
+    // both the tokenOwner (PublicKey) and parentTokenId
+    const accountId = {
+      tokenOwner,
+      parentTokenId
+    };
+    
+    // Create input similar to to_input(owner) in OCaml
+    const input: GenericHashInput<Field> = {
+      fields: [...PublicKey.toFields(accountId.tokenOwner), accountId.parentTokenId]
+    };
+    
+    // First pack the input (Random_oracle.pack_input), then hash with the prefix
+    const packed = packToFields(input);
+    const hash = hashWithPrefix(prefixes.deriveTokenId, packed) as Field;
+    
+    return new TokenId(hash);
+  }
+
+  /**
+   * Creates a TokenId from a Field value.
+   * @param field - The Field value.
+   * @returns A new TokenId.
+   */
+  static fromField(field: Field): TokenId {
```

Comment:
I don't see additional value to the original constructor here `constructor(public value: Field) {}`